poptorch_experimental_addons/cpp/replicatedallreducetp.cpp

```diff
@@ -0,0 +1,188 @@
+// cppimport
+// NOTE: the cppimport comment is necessary for dynamic compilation when loading
+// Copyright (c) 2022 Graphcore Ltd. All rights reserved.
+
+/*
+For the op to work you need to run the `OpToIdentityPattern` after autodiffing the op.
+*/
+
+#include <map>
+#include <memory>
+#include <snap/Tensor.hpp>
+#include <vector>
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsign-compare"
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#include <popart/alias/aliasmodel.hpp>
+#include <popart/basicoptionals.hpp>
+#include <popart/error.hpp>
+#include <popart/graph.hpp>
+#include <popart/ir.hpp>
+#include <popart/op.hpp>
+#include <popart/opmanager.hpp>
+#include <popart/opserialiser.hpp>
+#include <popart/popx/devicex.hpp>
+#include <popart/popx/irlowering.hpp>
+#include <popart/popx/opxmanager.hpp>
+#include <popart/popx/popopx.hpp>
+#include <popart/region.hpp>
+#include <popart/tensor.hpp>
+#include <popart/util.hpp>
+#include <popart/op/collectives/collectives.hpp>
+#include <popart/op/collectives/replicatedallreduce.hpp>
+#include <popart/popx/op/collectives/replicatedallreducex.hpp>
+#pragma GCC diagnostic pop
+
+
+#include <poplar/Graph.hpp>
+#include <poputil/exceptions.hpp>
+
+#include <gcl/Collectives.hpp>
+
+using InMapType = std::map<popart::InIndex, popart::TensorId>;
+using OutMapType = std::map<popart::OutIndex, popart::TensorId>;
+using OutIndex = int;
+
+namespace popart
+{
+  // -------------- Op --------------
```

Comment:
Sorry, we don't have CI for this yet, but could you run `clang-format` on this file to standardise it?